scripts/package

```diff
@@ -13,16 +13,67 @@ fi
 
 cp -r bin package/
 
-APIVERSION=`./bin/longhorn-share-manager version --client-only|jq ".clientVersion.apiVersion"`
+#APIVERSION=`./bin/longhorn-share-manager version --client-only|jq ".clientVersion.apiVersion"`
+
+# Try bin/longhorn-share-manager-amd64 and bin/longhorn-share-manager-arm64 doesn't have "Exec format error"
+# if it is not "Exec format error", use it to get the version
+#APIVERSION=`./bin/longhorn-share-manager version --client-only|jq ".clientVersion.apiVersion"`
+
+APIVERSION=""
+arch=$(uname -m)
+if [ "$arch" == "aarch64" ]; then
+    ARCH="arm64"
+else
+    ARCH="amd64"
+fi
+APIVERSION=`./bin/longhorn-share-manager-"$ARCH" version --client-only|jq ".clientVersion.apiVersion"`
+
+
 TAG=${TAG:-"v${APIVERSION}_`date -u +%Y%m%d`"}
 REPO=${REPO:-longhornio}
 IMAGE=${REPO}/${PROJECT}:${TAG}
 
+function convert_version_to_major_minor_x() {
+    local version="$1"
+    if [[ "$version" =~ ^v([0-9]+)\.([0-9]+)\. ]]; then
+        echo "v${BASH_REMATCH[1]}.${BASH_REMATCH[2]}.x"
+    else
+        echo "Invalid version format: $version"
+    fi
+}
+
+function get_branch() {
+  local version_file="version"
+  if [[ ! -f $version_file ]]; then
+    echo "Error: Version file '$version_file' not found."
+    exit 1
+  fi
+
+  local version=$(cat "$version_file")
+  local branch=$(convert_version_to_major_minor_x "$version")
+
+  # Fetch versions.json from the appropriate branch, fallback to main
+  wget -q "https://raw.githubusercontent.com/longhorn/dep-versions/${branch}/versions.json" -O /versions.json
```

Comment:
Will we use the downloaded `versions.json`? I thought dep-versions is cloned in the building container when running `buildx build`, so why still need this file? or just want to check the remote branch exists?

Reply:
No, this is used to determine the branch (master or v1.8.x, ..), and we don't really need to `version.json`.